modules/apps/transfer/types/trace.go

```diff
@@ -32,9 +37,25 @@ func ParseDenomTrace(rawDenom string) DenomTrace {
 		}
 	}
 
+	path := []string{}
+	baseDenom := []string{}
+	length := len(denomSplit)
+	r, err := regexp.Compile(fmt.Sprintf("%s[0-9]+", channeltypes.ChannelPrefix))
+	if err != nil {
+		panic("could not compile regexp 'channel-[0-9]+'")
```

Comment:
My understanding is that we recover from the panics higher up in the framework, but if we can now fail parsing the string, maybe it makes sense to add an `error` as a return value (it looks like the calling methods already return an error)

If you would prefer to leave the panic in, we could use the regexp.MustCompile option which already panics.

Reply:
Thanks for the suggestion, @chatton! Since Aditya recommends to remove the regex for channel identifier, then we will not need this anymore. :)